### PR TITLE
Refactor(server): prisma schema service

### PR DESCRIPTION
**Pull request by abrl91**
Part of: #6437 

### Summary
🗺️🧪🔧

This pull request adds and updates several files in the `prismaSchemaUtils` module. The main purpose is to support the conversion of Prisma schema to Amplication entities and fields, and to allow the upload and processing of Prisma schema files. The changes include adding helper and validator functions, updating unit tests, and adding constants for id type properties.

> _To convert Prisma schema to Amplication_
> _We need some utils for the transformation_
> _We add some helpers and validators_
> _And some objects and constants_
> _And update the tests for the `schema-utils` function_

### Walkthrough
*  Add helper functions for formatting and transforming Prisma schema elements in `helpers.ts` ([link](https://github.com/amplication/amplication/pull/6490/files?diff=unified&w=0#diff-eda55e374cb3117966721fdd6185983e4ceade02d22ad39b39aee47863e36e58R1-R211))
* Add validator functions for validating Prisma schema against Prisma and Amplication rules in `validators.ts` ([link](https://github.com/amplication/amplication/pull/6490/files?diff=unified&w=0#diff-5b3d89f93345ce6b2357985bc61579b8a092d7c33b72bb9363b8cee47c7a4f2aR1-R44))
* Add unit tests for helper functions in `helpers.spec.ts` ([link](https://github.com/amplication/amplication/pull/6490/files?diff=unified&w=0#diff-551ae1847949455e1024d78b28dcd5875c092f40a2e1d07ebbe0b588c84c9174R1-R91))
* Add id type property mappings to constants in `constants.ts` ([link](https://github.com/amplication/amplication/pull/6490/files?diff=unified&w=0#diff-b5be81781d293f79ac5277d34d4c5556e3401b3159d7d89722e68d518248b36bR24-R36))
* Update import statements and remove duplicate tests in `schema-utils.spec.ts` ([link](https://github.com/amplication/amplication/pull/6490/files?diff=unified&w=0#diff-24c7617f42d49e0aee9865b33757c9056e2380480b904240fea405dbe0f0d450L2-R12), [link](https://github.com/amplication/amplication/pull/6490/files?diff=unified&w=0#diff-24c7617f42d49e0aee9865b33757c9056e2380480b904240fea405dbe0f0d450L23-R209))

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/prismaSchemaUtils/helpers.ts
Comment (mulygottlieb) [inline packages/amplication-server/src/core/prismaSchemaUtils/helpers.ts:1]: A lot of the functions here have a `return` statement only under an `if` condition, meaning that it returns undefined in other cases. This is implicit and may be confusing for the caller. What do you expect those functions to return if the `if` condition is not met? Should we add tests for those cases?

File: packages/amplication-server/src/core/prismaSchemaUtils/validators.ts [hunk 1]
```diff
@@ -13,11 +13,8 @@ export function validateSchemaUpload(file: string): void {
   const schemaString = file.replace(/\\n/g, "\n");
   try {
     validate({ datamodel: schemaString });
-    this.logger.info("Valid schema");
   } catch (error) {
-    this.logger.error(error.message, error, {
-      functionName: "validateSchemaUpload",
-    });
+    console.error(error);
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/prismaSchemaUtils/validators.ts:18]: Do we really want a `console.error()` here and not any logger?

Comment (mulygottlieb) [reply packages/amplication-server/src/core/prismaSchemaUtils/validators.ts:18]: Then you will also not need the `eslint-disable-next-line` that we're trying to reduce the use of anyway

Comment (abrl91) [reply packages/amplication-server/src/core/prismaSchemaUtils/validators.ts:18]: Yes. The console error shows the error string from Prisma. It is not a must but it is nice to have it

File: packages/amplication-server/src/core/prismaSchemaUtils/helpers.ts
Comment (abrl91) [reply packages/amplication-server/src/core/prismaSchemaUtils/helpers.ts:1]: All functions here are not new. I just copied them from the schema-utils file

Comment (abrl91) [reply packages/amplication-server/src/core/prismaSchemaUtils/helpers.ts:1]: If you are talking about the functions that checks the types, it should return only if the type is what we are checking

**Review by mulygottlieb — APPROVED** [review]

**Pull request merged**